tarodb/assets_store.go

```diff
@@ -1556,6 +1562,43 @@ func (a *AssetStore) LogPendingParcel(ctx context.Context,
 	})
 }
 
+// LogParcelDeliveryAttempt logs a parcel delivery attempt to disk.
+func (a *AssetStore) LogParcelDeliveryAttempt(ctx context.Context,
+	anchorPoint wire.OutPoint) error {
+
+	anchorPointBytes, err := encodeOutpoint(anchorPoint)
+	if err != nil {
+		return err
+	}
+
+	var writeTxOpts AssetStoreTxOptions
+	return a.db.ExecTx(ctx, &writeTxOpts, func(q ActiveAssetsStore) error {
+		// First, we'll fetch the asset transfer using the given
+		// outpoint bytes. We can then relate the delivery attempt to
+		// the asset transfer entry.
+		assetTransfers, err := q.QueryAssetTransfers(ctx, TransferQuery{
+			NewAnchorPoint: anchorPointBytes,
+		})
+		if err != nil {
+			return err
```

Comment:
We should try to always wrap the error with some additional info to make debugging easier (e.g. what query failed exactly?). So for example: `return fmt.Errorf("unable to query asset transfers: %w", err)`.